pkgs/os-specific/linux/wpa_supplicant/unprivileged-daemon.patch

```diff
@@ -0,0 +1,96 @@
+commit 24e932357ee3041763135b931206dfc0bbe0441e
+Author: rnhmjoj <rnhmjoj@inventati.org>
+Date:   Wed Jul 23 10:18:55 2025 +0200
+
+    Fixes for running wpa_supplicant unprivileged
+    
+    1. Change the dbus service user to "wpa_supplicant"
+    
+    2. Ensure appropriate group ownership and permissions on the client sockets.
+       Motivation: clients communicate with the daemon by creating "client"
+       sockets; by default this is owned by the user running the client,
+       so it may be inaccessible by the daemon.
+    
+    3. Move the "control" sockets under a subdirectory of /run/wpa_supplicant.
+       Motivation: wpa_supplicant will try to adjust the ownership of the
+       sockets directory, even if they are fine, and fail.
+    
+    4. Move the "client" under a subdirectory of /run/wpa_supplicant instead
+       of tmp. Motivation: this allows to unshare /tmp
+
+diff --git a/src/common/wpa_ctrl.c b/src/common/wpa_ctrl.c
+index 7e197f0..6bfb091 100644
+--- a/src/common/wpa_ctrl.c
++++ b/src/common/wpa_ctrl.c
+@@ -15,6 +15,8 @@
+ #include <fcntl.h>
+ #include <sys/un.h>
+ #include <unistd.h>
++#include <sys/types.h>
++#include <grp.h>
+ #include <fcntl.h>
+ #endif /* CONFIG_CTRL_IFACE_UNIX */
+ #ifdef CONFIG_CTRL_IFACE_UDP_REMOTE
+@@ -165,6 +167,14 @@ try_again:
+ 		return NULL;
+ 	}
+ 
++	/* Set the client socket owner group to "wpa_supplicant"
++	 * and ensure group and user permissions are the same */
++	struct group *grp = getgrnam("wpa_supplicant");
++	if (grp != NULL) {
++		lchown(ctrl->local.sun_path, -1, grp->gr_gid);
```

Comment:
Shouldn’t the socket be owned by the creating user and group by default?

Not doing this would avoid having to hard‐code the `wpa_supplicant` group name.

Changing permissions however makes sense in my opinion considering the umask.

Reply:
> Shouldn’t the socket be owned by the creating user and group by default?

No, that's in fact what `wpa_cli` would do without the patch. It doesn't work because then the `wpa_supplicant` daemon can't access the socket.

Reply:
Correct,